app/src/main/java/ai/elimu/appstore/util/VersionHelper.java

```diff
@@ -37,12 +39,25 @@ public static void updateAppVersion (@NonNull Context context) {
             AppPrefs.saveAppVersionCode(newVersionCode);
             oldVersionCode = newVersionCode;
         }
+        Timber.i("oldVersionCode: " + oldVersionCode);
+        Timber.i("newVersionCode: " + newVersionCode);
+
+        // Handle upgrade from previous version
         if (oldVersionCode < newVersionCode) {
-            Timber.i("Upgrading application from version " + oldVersionCode + " to " +
-                    newVersionCode);
-//            if (newVersionCode == ???) {
-//                // Put relevant tasks required for upgrading here
+            Timber.i("Upgrading application from version " + oldVersionCode + " to " + newVersionCode + "...");
+
+            if (oldVersionCode < 2000011) {
```

Comment:
This will basically delete all APKs whenever there's app version upgrade, regardless of having database version upgraded or not. Is that what we are expecting?

Reply:
No, only if the previously installed application's versionCode is lower than 2.0.11. Note that if the app is installed for the first time, `oldVersionCode` is set to be equal to the `newVersionCode`.

Reply:
The upgrade code is only triggered if there was an app version upgrade, yes. It does not have anything to do with the SQLite database.

Reply:
Okay I understand now.